vocabulary manager broken when creating first vocab in a school.

borkage happens _after_ the vocab is saved to the server.

must have something to do with the expand/collapse toggle, and the request param that comes with that.

```
error.js:28 Error: Assertion Failed: calling set on destroyed object: <ilios@component:school-vocabularies-list::ember1076>.showNewVocabularyForm = [object Boolean]
    at new Error (native)
    at Error.EmberError (http://localhost:4200/assets/vendor.js:30207:21)
    at assert (http://localhost:4200/assets/vendor.js:21049:13)
    at Object.assert (http://localhost:4200/assets/vendor.js:29971:34)
    at Object.set (http://localhost:4200/assets/vendor.js:34576:22)
    at Class.set (http://localhost:4200/assets/vendor.js:49277:38)
    at http://localhost:4200/assets/ilios.js:11265:17
    at tryCatch (http://localhost:4200/assets/vendor.js:65999:14)
    at invokeCallback (http://localhost:4200/assets/vendor.js:66014:15)
    at publish (http://localhost:4200/assets/vendor.js:65982:9)
```

--- a/app/components/school-vocabularies-list.js
+++ b/app/components/school-vocabularies-list.js
@@ -47,15 +47,18 @@ export default Component.extend(Validations, ValidationErrorDisplay, {
       this.set('showNewVocabularyForm', !this.get('showNewVocabularyForm'));
     },
     add(title){
+      let self = this;
       this.set('isSavingNewVocabulary', true);
       const school = this.get('school');
       let vocabulary = this.get('store').createRecord('vocabulary', {title, school});
       vocabulary.save().then(vocabulary => {
-        this.send('removeErrorDisplayFor', 'newVocabularyTitle');
-        this.get('newVocabularies').pushObject(vocabulary);
-        this.set('showNewVocabularyForm', false);
-        this.set('isSavingNewVocabulary', false);
-        this.set('newVocabularyTitle', null);
+        if (!self.get('isDestroyed')) {
+          self.send('removeErrorDisplayFor', 'newVocabularyTitle');
+          self.get('newVocabularies').pushObject(vocabulary);
+          self.set('showNewVocabularyForm', false);
+          self.set('isSavingNewVocabulary', false);
+          self.set('newVocabularyTitle', null);
+        }
       });
     },
     confirmRemoval(vocabulary){

--- a/app/components/school-vocabularies-list.js
+++ b/app/components/school-vocabularies-list.js
@@ -49,15 +49,21 @@ export default Component.extend(Validations, ValidationErrorDisplay, {
     add(title){
       let self = this;
       this.set('isSavingNewVocabulary', true);
-      const school = this.get('school');
-      let vocabulary = this.get('store').createRecord('vocabulary', {title, school});
-      vocabulary.save().then(vocabulary => {
-        if (!self.get('isDestroyed')) {
-          self.send('removeErrorDisplayFor', 'newVocabularyTitle');
-          self.get('newVocabularies').pushObject(vocabulary);
-          self.set('showNewVocabularyForm', false);
-          self.set('isSavingNewVocabulary', false);
-          self.set('newVocabularyTitle', null);
+      this.send('addErrorDisplayFor', 'newVocabularyTitle');
+      this.validate().then(({validations}) => {
+        if (validations.get('isValid')) {
+          const school = this.get('school');
+          let vocabulary = this.get('store').createRecord('vocabulary', {title, school});
+          vocabulary.save().then(vocabulary => {
+            this.get('newVocabularies').pushObject(vocabulary);
+          }).finally(() => {
+            if (!self.get('isDestroyed')) {
+              self.send('removeErrorDisplayFor', 'newVocabularyTitle');
+              self.set('showNewVocabularyForm', false);
+              self.set('isSavingNewVocabulary', false);
+              self.set('newVocabularyTitle', null);
+            }
+          });
         }
       });
     },

